scripts/core/editor3/actions/toolbar.tsx

```diff
@@ -66,13 +67,36 @@ export function removeFormat() {
  */
 export function insertMedia(files?, targetBlockKey = null) {
     const superdesk = ng.get('superdesk');
+    const renditions = ng.get('renditions');
+
+    const editedMedia = [];
 
     return (dispatch) => {
-        superdesk.intent('upload', 'media', files).then((media) => {
-            dispatch({
-                type: 'TOOLBAR_INSERT_MEDIA',
-                payload: {files: media, targetBlockKey: targetBlockKey},
-            });
+        superdesk.intent('upload', 'media', files).then((media_list) => {
+            function editNext() {
+                if (media_list && Array.isArray(media_list) && media_list.length > 0) {
+                    const _media = media_list.shift();
+                    const _isImage = checkRenditions.isImage(_media.renditions.original);
+                    const options = {
+                        isNew: true,
+                        editable: true,
+                        defaultTab: _isImage ? 'crop' : 'view',
+                    };
+
+                    setTimeout(() => {
```

Comment:
why do you need `setTimeout`?

Reply:
I'm using this to let the current synchronous flow get over so that already opened modal gets closed and the new modal can be opened.

Reply:
:+1: add this as a comment in this code